framework/src/dgkernels/DGKernelWarehouse.C

```diff
@@ -0,0 +1,123 @@
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*           (c) 2010 Battelle Energy Alliance, LLC             */
+/*                   ALL RIGHTS RESERVED                        */
+/*                                                              */
+/*          Prepared by Battelle Energy Alliance, LLC           */
+/*            Under Contract No. DE-AC07-05ID14517              */
+/*            With the U. S. Department of Energy               */
+/*                                                              */
+/*            See COPYRIGHT for full restrictions               */
+/****************************************************************/
+
+#include "DGKernelWarehouse.h"
+#include "DGKernel.h"
+
+DGKernelWarehouse::DGKernelWarehouse() :
+    Warehouse<DGKernel>()
+{
+}
+
+DGKernelWarehouse::~DGKernelWarehouse()
+{
+}
+
+void
+DGKernelWarehouse::initialSetup()
+{
+  for (std::vector<DGKernel *>::const_iterator i = all().begin(); i != all().end(); ++i)
+    (*i)->initialSetup();
+}
+
+void
+DGKernelWarehouse::timestepSetup()
+{
+  for (std::vector<DGKernel *>::const_iterator i = all().begin(); i != all().end(); ++i)
+    (*i)->timestepSetup();
+}
+
+void
+DGKernelWarehouse::residualSetup()
+{
+  for (std::vector<DGKernel *>::const_iterator i = all().begin(); i != all().end(); ++i)
+    (*i)->residualSetup();
+}
+
+void
+DGKernelWarehouse::jacobianSetup()
+{
+  for (std::vector<DGKernel *>::const_iterator i = all().begin(); i != all().end(); ++i)
+    (*i)->jacobianSetup();
+}
+
+void
+DGKernelWarehouse::addDGKernel(MooseSharedPointer<DGKernel> & dg_kernel, const std::set<SubdomainID> & block_ids)
+{
+  _all_ptrs.push_back(dg_kernel);
+  _all_objects.push_back(dg_kernel.get());
+
+  // Check to make sure not boundary restricted
+  if (!dg_kernel->boundaryRestricted())
+  {
+    // Non-block restricted
+    if (block_ids.empty() || block_ids.find(Moose::ANY_BLOCK_ID) != block_ids.end())
+    {
+      _global_dg_kernels.push_back(dg_kernel.get());
+    }
+
+    // Block restricted
+    else
+    {
+      for (std::set<SubdomainID>::iterator it = block_ids.begin(); it != block_ids.end(); ++it)
+      {
+        SubdomainID blk_id = *it;
+        _block_dg_kernels[blk_id].push_back(dg_kernel.get());
+      }
+    }
+  }
+}
+
+void
+DGKernelWarehouse::addInterfaceDG(const std::set<BoundaryID> & boundary_ids, MooseSharedPointer<DGKernel> & interface_dg)
+{
+  for (std::set<BoundaryID>::const_iterator it = boundary_ids.begin(); it != boundary_ids.end(); ++it)
+    _interface_dg_kernels[*it].push_back(interface_dg.get());
+}
+
+void
+DGKernelWarehouse::updateActiveDGKernels(Real /*t*/, Real /*dt*/, unsigned int subdomain_id)
+{
+  _active_internal_dg_kernels.clear();
+  _active_var_dg_kernels.clear();
+
+    // FIXME: add startTime/stopTime to DGKernel
+//    if (dg_kernel->startTime() <= t + (1e-6 * dt) && dg_kernel->stopTime() >= t + (1e-6 * dt))
```

Comment:
I still want to check this.  Cloning it now...

On Mon, Jan 11, 2016 at 12:18 PM David Andrs <notifications@github.com>
wrote:

> In framework/src/dgkernels/DGKernelWarehouse.C
> <https://github.com/idaholab/moose/pull/5943#discussion_r49348674>:
>
> > +
> > +void
> > +DGKernelWarehouse::addInterfaceDG(const std::set<BoundaryID> & boundary_ids, MooseSharedPointer<DGKernel> & interface_dg)
> > +{
> > +  for (std::set<BoundaryID>::const_iterator it = boundary_ids.begin(); it != boundary_ids.end(); ++it)
> > +    _interface_dg_kernels[*it].push_back(interface_dg.get());
> > +}
> > +
> > +void
> > +DGKernelWarehouse::updateActiveDGKernels(Real /*t*/, Real /*dt*/, unsigned int subdomain_id)
> > +{
> > +  _active_internal_dg_kernels.clear();
> > +  _active_var_dg_kernels.clear();
> > +
> > +    // FIXME: add startTime/stopTime to DGKernel
> > +//    if (dg_kernel->startTime() <= t + (1e-6 * dt) && dg_kernel->stopTime() >= t + (1e-6 * dt))
>
> I assume, we do not want this line here...
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/idaholab/moose/pull/5943/files#r49348674>.
>